addons/arsenal/functions/fnc_fillRightPanel.sqf

```diff
@@ -0,0 +1,331 @@
+        {
+            ["CfgMagazines", _x, true, true] call _fnc_fill_right_Container;
+        } foreach ((GVAR(virtualItems) select 19) arrayIntersect ((_compatibleMagazines select 0) + (_compatibleMagazines select 1) + (_compatibleMagazines select 2)));
+    };
+
+    case IDC_buttonMagALL : {
+        {
+            ["CfgMagazines", _x, true] call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 2);
+        {
+            ["CfgMagazines", _x, true, true]  call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 19);
+    };
+
+    case IDC_buttonThrow : {
+        {
+            ["CfgMagazines", _x, true]  call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 15);
+        {
+            ["CfgMagazines", _x, true, true]  call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 20);
+    };
+
+    case IDC_buttonPut : {
+        {
+            ["CfgMagazines", _x, true]  call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 16);
+        {
+            ["CfgMagazines", _x, true, true]  call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 21);
+    };
+
+    case IDC_buttonMisc : {
+        {
+            ["CfgWeapons", _x, false]  call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 17);
+        {
+            ["CfgWeapons", _x, false, true]  call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 18);
+        {
+            ["CfgVehicles", _x, false, true]  call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 23);
+        {
+            ["CfgGlasses", _x, false, true]  call _fnc_fill_right_Container;
+        } foreach (GVAR(virtualItems) select 24);
+    };
+};
+
+if (GVAR(currentRightPanel) != _ctrlIDC) then {
+    (_display displayCtrl IDC_rightSearchbar) ctrlSetText "";
+};
+
+GVAR(currentRightPanel) = _ctrlIDC;
+
+[QGVAR(rightPanelFilled), [_display, _ctrlIDC]] call CBA_fnc_localEvent;
+
+// Sorting
+private _sortRightCtrl = _display displayCtrl IDC_sortRightTab;
+
+if (lbSize _sortRightCtrl == 3) then {
+    _sortRightCtrl lbDelete 2;
+};
+
+if (_leftPanelState) then {
+    _sortRightCtrl lbDelete 1;
+    _sortRightCtrl lbAdd (localize "STR_a3_rscdisplayarsenal_sort_mod");
+    _sortRightCtrl lbSetValue [1, 1];
+} else {
+
+    _sortRightCtrl lbDelete 1;
+    _sortRightCtrl lbAdd localize LSTRING(sortByWeightText);
+    _sortRightCtrl lbSetValue [1, 1];
+    
+    _sortRightCtrl lbAdd localize LSTRING(sortByAmountText);
+    _sortRightCtrl lbSetValue [2, 2];
+};
+
+[_sortRightCtrl, _sortRightCtrl lbValue (lbCurSel _sortRightCtrl)] call FUNC(sortPanel);
+
+// Add current items and change progress bar
+if (GVAR(currentLeftPanel) in [IDC_buttonUniform, IDC_buttonVest, IDC_buttonBackpack]) then {
+    private _container = switch (GVAR(currentLeftPanel)) do {
```

Comment:
switch statement can be replaced by ARRAY find and an array of CODE and a `call`.
_container = call (CODEARRAY select (ARRAY find CURPanel);
That's 3 script commands. 
vs 2+2+2+2 -> 8 for the switch.